featuretools/tests/entityset_tests/test_es.py

```diff
@@ -89,10 +89,12 @@ def test_reset_metadata(es):
 
 
 def test_cannot_re_add_relationships_that_already_exists(es):
-    before_len = len(es.relationships)
-    es.add_relationship(es.relationships[0])
-    after_len = len(es.relationships)
-    assert before_len == after_len
+    warn_text = "Not adding duplicate relationship: " + str(es.relationships[0])
+    with pytest.warns(UserWarning, match=warn_text):
+        before_len = len(es.relationships)
```

Comment:
I think we should limit the code run in the `with pytest.warns(..):` block to just the line we expect to trigger the warning.  In this case, `es.add_relationship`. 

We should also do this in `test_none_index` and `test_unknown_index`